lib/api.ts

```diff
@@ -0,0 +1,283 @@
+import marked, { Tokens } from "marked";
+
+import { DocsTree } from "~/components/docs-navigation";
+import { addHrefPrefix, urlToSlugs } from "~/helpers/params";
+import {
+  isCollaborator,
+  isHeading,
+  isLink,
+  isList,
+  isListItemWithTokens,
+  isString,
+  isTextWithTokens,
+} from "~/helpers/type-guards";
+import { GithubCollaborator } from "~/types/api";
+
+const customOctokit = Octokit.plugin(paginateRest);
+
+const octokit = new customOctokit({
+  auth: `${process.env.GITHUB_TOKEN}`,
+});
+
+export const OWNER = "strawberry-graphql";
+export const REPO = "strawberry";
+// default branch to query against.
+export const REF = "master";
+/**
+ * Contributor username/login to ignore as they are bots.
+ */
+const IGNORE_LIST = ["dependabot-preview[bot]", "dependabot-bot", "botberry"];
+
+/**
+ * fetchContributors uses the GitHub REST API for contributors as the GraphQL
+ * API auth token needs push access. @octokit/plugin-paginate-rest gives us
+ * support for paginatated responses. So if we ever get above 100 contributors
+ * this should handle querying more pages for us.
+ */
+export const fetchContributors: () => Promise<
+  GithubCollaborator[]
+> = async () => {
+  const contributors: string[] = await octokit.paginate(
+    "GET /repos/{owner}/{repo}/contributors",
+    { owner: OWNER, repo: REPO, per_page: 100 },
+    (response) =>
+      response.data
+        .filter((member) =>
+          isString(member.login) ? !IGNORE_LIST.includes(member.login) : false
+        )
+        .map((member) => member.login)
+        .filter(isString)
+  );
+
+  /**
+   * Get a contributors name & blog. If the request errors then return undefined
+   *  and filter item.
+   */
+  return Promise.all(
+    contributors.map(async (username) => {
+      try {
+        const { data } = await octokit.request("GET /users/{username}", {
+          username,
+        });
+
+        return {
+          name: data.name,
+          login: data.login,
+          url: data.blog ?? data.html_url,
+        };
+      } catch {
+        return;
+      }
+    })
+  ).then((resolve) => resolve.filter(isCollaborator));
+};
+
+export const fetchPullRequest = async ({
+  pull_number,
+  owner = OWNER,
+  repo = REPO,
+}: {
+  pull_number: number;
+  owner?: string;
+  repo?: string;
+}): Promise<{
+  pull_number: number;
+  owner: string;
+  repo: string;
+  branch: string;
+  html_url: string;
+}> => {
+  try {
+    const pull = await octokit.request(
+      "GET /repos/{owner}/{repo}/pulls/{pull_number}",
+      { owner, repo, pull_number }
+    );
+
+    const safeToPreview: boolean =
+      pull.data.state === "open" &&
+      pull.data.labels.find((label) => label.name === "ok-to-preview") !==
+        undefined;
+
+    if (!safeToPreview) {
+      // TODO: we might want to show a message to the users that a pull request needs the label of "ok-to-preview" instead of throwing an error which 404s.
+      throw new Error("not safe to preview");
+    }
```

Comment:
yes, this could be a good idea, for now the links are only visible when adding the label, so we can keep it like that :)